### DISCUSSION TITLE
Lesson 6: No Value error is coming for mainnet-fork

### DISCUSSION THREAD

**Discussion by gauravupadhyay1994** [smartcontractkit/full-blockchain-solidity-course-py#1151] [Q&A] [answered]
when I execute, brownie run scripts/deploy.py --network mainnet-fork, it gets executed successfully. No ValueError is coming as in the video. 

PS D:\demos\brownie_fund_me> brownie run scripts/deploy.py --network mainnet-fork
INFO: Could not find files for the given pattern(s).
Brownie v1.16.4 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Launching 'ganache-cli.cmd --accounts 10 --hardfork istanbul --fork https://mainnet.infura.io/v3/69690316fe1d475c93b12647xxxxxxx --gasLimit 12000000 --mnemonic brownie --port 8545 --chainId 1'...

Running 'scripts\deploy.py::main'...
Transaction sent: 0xfbd5c37141c91389e80e95cf1ea0b5af82132ba5b87efd54bfa6f9b73644bcc4
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 0
  FundMe.constructor confirmed   Block: 14276780   Gas used: 405741 (3.38%)
  FundMe deployed at: 0x350A16e2d2f382e0C5c885Af6cED8F8611a34Bea

Contract deployed to 0x350A16e2d2f382e0C5c885Af6cED8F8611a34Bea
Terminating local RPC client...

**Comment by n4n0b1t3** [comment]
Dude, are you complaining, that you are NOT getting an error? That would be a new one :-)
No, seriously, I believe the terminal output you posted is not complete, because I can't see an error.
Can you please adapt your post, and [have a look at this article here](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/980). It helps us to help you when the post has a certain format.

**Reply by cromewar** [reply]
Hello @GaUrAvUp, yes I have a barely idea about what this is related to, but is really hard to know with certainty if you don't provide more info on this, please follow @n4n0b1t3  instructions.

**Reply by gauravupadhyay1994** [reply]
I have been following the video exactly the way it is being taught. At 05:57:24, when I deployed my code, it should have shown some ValueError because I had no gas to spend on mainnet-fork at infura.io.  My code matches exactly with Patrick's code. 
![config_fundMe](https://user-images.githubusercontent.com/11389063/155868445-85f99379-256b-4053-9ead-d0c2e8aa2279.png)

**Reply by alexkalen** [reply]
I have the same issue so Im going to complete the original post. In 5:59 Patrick comes to an error when deploying the mainnet-fork network, he says that the error is due to brownies built-in forking mechanism not creating the accounts (so there is no gas to spend on transaction fees). But whenever I run the same script I have no issues. So no error shows up. So should we just move on? I guess that brownie is creating the acounts needed for the mainnet-fork. 

![image](https://user-images.githubusercontent.com/24979100/169085177-83681c87-a5cd-4347-912e-5d17171cdc94.png)

**Reply by alexkalen** [reply]
The arent any atual errors. As yu can see. 

```C:\Users\user\demos\brownie_fund_me> brownie run scripts/deploy.py --network mainnet-fork
INFO: Could not find files for the given pattern(s).
Brownie v1.18.1 - Python development framework for Ethereum

BrownieFundMeProject is the active project.
C:\Users\alexk\.local\pipx\venvs\eth-brownie\lib\site-packages\brownie\network\main.py:44: BrownieEnvironmentWarning: Development network has a block height of 11
  warnings.warn(
Attached to local RPC client listening at '127.0.0.1:8545'...

Running '\Users\alexk\demos\brownie_fund_me\scripts\deploy.py::main'...
Transaction sent: 0x2b6cf887fad2b88e8d5ef6462b535f63bf4db069d8bda9b260bd542061dd43d9
  Gas price: 0.0 gwei   Gas limit: 6721975   Nonce: 8
  FundMe.constructor confirmed   Block: 12   Gas used: 405957 (6.04%)
  FundMe deployed at: 0xdAD3700b4Bc3AdA70cA3a9F42293c41E10Fe4b0c

Contract deployed to 0xdAD3700b4Bc3AdA70cA3a9F42293c41E10Fe4b0c
PS C:\Users\alexk\demos\brownie_fund_me>```

**Comment by n4n0b1t3** [comment, marked as answer]
The array out of range error at this timestamp stems from the fact, that the brownie version Patrick uses in this video, does not automatically create accounts for mainnet-fork. 
Your ganache-cli output shows, however, that 10 accounts are created
`Launching 'ganache-cli.cmd --accounts 10 --hardfork istanbul --fork https://mainnet.infura.io/v3/69690316fe1d475c93b12647xxxxxxx --gasLimit 12000000 --mnemonic brownie --port 8545 --chainId 1'...`
Another thing I noticed, is that you are using mainnet-fork but have configured mainnet-fork-dev in your config yaml. This could cause some errors ahead.

**Reply by cromewar** [reply]
Yeap @n4n0b1t3 is right your `brownie-config.yaml` has `mainnet-fork-dev` and you are deploying using just `mainnet-fork`

**Reply by MananJain2002** [reply]
HI the only reason this is happening is because our mainnet-fork is running on Ganache-CLI unlike for patrick's which is running on different environment
![Ganache-CLI](https://user-images.githubusercontent.com/63538502/165140329-1839ea34-2e12-458e-8766-3d7b8af9aaf6.png)
![mainnet-fork](https://user-images.githubusercontent.com/63538502/165140339-84f914be-5f80-4cca-babf-ea284e642531.png)

**Reply by alexkalen** [reply]
Alright, this makes sense. So if anyone encounters this same "issue" should we just move on to Lessonn 7 and not create the mainnet-fork-dev?
